code/modules/food&drinks/kitchen machinery/food_cart.dm

```diff
@@ -0,0 +1,147 @@
+	dat += "</div><br><b>MIXER CONTENTS</b><br><div class='statusDisplay'>"
+	for(var/datum/reagent/R in mixer.reagents.reagent_list)
+		dat += "[R.name]: [R.volume] "
+		dat += "<a href='?src=\ref[src];transfer=[R.id]'>Transfer back</a>"
+		if (glasses > 0)
+			dat += "<a href='?src=\ref[src];m_pour=[R.id]'>Pour in a glass</a>"
+		dat += "<br>"
+	dat += "</div><br><b>STORED FOOD</b><br><div class='statusDisplay'>"
+	for(var/V in stored_food)
+		if(stored_food[V] > 0)
+			dat += "<b>[V]: [stored_food[V]]</b> <a href='?src=\ref[src];dispense=[V]'>Dispense</a><br>"
+	dat += "</div><br><a href='?src=\ref[src];refresh=1'>Refresh</a> <a href='?src=\ref[src];close=1'>Close</a>"
+
+	var/datum/browser/popup = new(user, "foodcart","Food Cart", 500, 350, src)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/food_cart/proc/isFull()
+	return food_stored >= STORAGE_CAPACITY
+
+/obj/machinery/food_cart/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	if(istype(O, /obj/item/weapon/reagent_containers/food/drinks/drinkingglass))
+		var/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/DG = O
+		if(!DG.reagents.total_volume) //glass is empty
+			user.drop_item()
+			qdel(DG)
+			glasses++
+			user << "<span class='notice'>The [src] accepts drinking glass, sterilizing it.</span>"
+	else if(istype(O, /obj/item/weapon/reagent_containers/food/snacks))
+		if(isFull())
+			user << "<span class='warning'>The [src] is at full capacity.</span>"
+		else
+			var/obj/item/weapon/reagent_containers/food/snacks/S = O
+			user.drop_item()
+			S.loc = src
+			if(stored_food[sanitize(S.name)])
+				stored_food[sanitize(S.name)]++
+			else
+				stored_food[sanitize(S.name)] = 1
+	else if(istype(O, /obj/item/stack/sheet/glass))
+		var/obj/item/stack/sheet/glass/G = O
+		if(G.get_amount() >= 1)
+			G.use(1)
+			glasses += 4
+			user << "<span class='notice'>The [src] accepts a sheet of glass.</span>"
+	else if(istype(O, /obj/item/weapon/storage/bag/tray))
+		var/obj/item/weapon/storage/bag/tray/T = O
+		for(var/obj/item/weapon/reagent_containers/food/snacks/S in T.contents)
+			if(isFull())
+				user << "<span class='warning'>The [src] is at full capacity.</span>"
+				break
+			else
+				T.remove_from_storage(S, src)
+				if(stored_food[sanitize(S.name)])
+					stored_food[sanitize(S.name)]++
+				else
+					stored_food[sanitize(S.name)] = 1
+	else if(O.is_open_container())
+		return
+	else
+		..()
+	updateDialog()
+
+/obj/machinery/food_cart/Topic(href, href_list)
+	if(..())
+		return
+
+	if(href_list["disposeI"])
+		reagents.del_reagent(href_list["disposeI"])
+
+	if(href_list["dispense"])
+		if(stored_food[href_list["dispense"]]-- <= 0)
+			stored_food[href_list["dispense"]] = 0
+		else
+			for(var/obj/O in contents)
+				if(sanitize(O.name) == href_list["dispense"])
+					O.loc = src.loc
+					break
+
+	if(href_list["portion"])
+		portion = max(0, min(50, input("How much drink do you want to dispense per glass?") as num))
```

Comment:
nitpick: You can use `Clamp(n, 0, 50)` instead of using max and min, it's more readable IMO